src/extension.ts

```diff
@@ -352,20 +360,27 @@ export async function activate(context: extensionApi.ExtensionContext): Promise<
         },
       )
       .then(() => false)
-      .catch(() => true); // required to force Promise.all() call keep waiting for all not failed calls
+      .catch((error) => {
+        return true; // required to force Promise.all() call keep waiting for all not failed calls
+      });
 
     const failed = await Promise.all([registryAccess, vmActivation]); // wait for all fail or pass
 
     if (failed.some(fail => fail)) {
       removeSession(currentSession.id); // if at least one fail, remove session
+      telemetryData.successful = false;
     }
+
+    TelemetryLogger.logUsage('signin', telemetryData);
```

Comment:
I think we're missing lot of data in telemetryData

here it's only successful yes/no

I would expect the time it has taken, in case of failure, which step succeed/failed.


did we have to restart and how many restart

at which step the user stopped the flow

for example it's different if user aborted if no podman machine was running vs error in doing the task
did we have a PodmanMachineRunning ?

was isSubscriptionManagerInstalled already installed ?

Was it registryAccess part, vmActivation part

for each step, the time it took as well.

Then about the session, probably need some data (about the user being authenticated
I don't know what data we have but maybe number about some fields of the user (counter)

Reply:
@benoitf @slemeur would be possible to fix all of that as follow up issues? I have added part of it and working on tests right now, but it will take some time to implement.

Reply:
I think that's fair.  We had a meeting on the telemetry issue (https://github.com/redhat-developer/podman-desktop-redhat-account-ext/issues/43#issuecomment-1997959842) where these details did not come up.  So I guess we under-specified the requirements initially.

Merging the PR and trying to get as far as possible until the deadline is OK for me.